File: model/src/main/java/com/testerum/model/manual/operation/UpdateManualTestModel.kt
package com.testerum.model.manual.operation

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.testerum.model.infrastructure.path.Path
import com.testerum.model.manual.ManualTest

data class UpdateManualTestModel @JsonCreator constructor(
        @JsonProperty("oldPath") val oldPath: Path,
        @JsonProperty("manualTest") val manualTest: ManualTest)

File: web/web-frontend/src/app/manual-tests/model/operation/update-manual-test.model.ts
import {JsonUtil} from "../../../utils/json.util";
import {ManualTestModel} from "../manual-test.model";
import {Path} from "../../../model/infrastructure/path/path.model";
import {Serializable} from "../../../model/infrastructure/serializable.model";

export class UpdateManualTestModel implements Serializable<UpdateManualTestModel> {

    oldPath:Path;
    manualTestModel:ManualTestModel;

    constructor(oldPath: Path, manualTestModel: ManualTestModel) {
        this.oldPath = oldPath;
        this.manualTestModel = manualTestModel;
    }

    deserialize(input: Object): UpdateManualTestModel {
        this.oldPath = Path.deserialize(input["oldPath"]);
        this.manualTestModel = input['manualTest'];

        return this;
    }

    serialize() {
        return "" +
            '{' +
            '"oldPath":' + JsonUtil.serializeSerializable(this.oldPath) + ',' +
            '"manualTest":' + JsonUtil.serializeSerializable(this.manualTestModel) +
            '}'
    }
}
